### PR TITLE
feat: [UX] Permalink to grid/list view

### PR DESCRIPTION
**Pull request by OronNadiv**
## Context
Users want to have the ability to go directory to grid/list view.

## Objective
Create a permalink to `pipeline.events.show` in "list" or "grid" view.

This permlink (`/pipelines/123/events/456?view=list`) will directly show the list view:
image

This permlink (`/pipelines/123/events/456?view=grid`) will directly show the grid view:
image

Clicking the toggle will change the URL (`view=grid` vs `view=list`) 

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by OronNadiv** [comment]
@adong I am aware that the tests are failing but wanted to have your input about my change to see that I'm addressing it correctly.  If my solution is right, then I'll go ahead and fix the tests.  Thanks.

**Review by jithine — COMMENTED** [review]
grid view shouldn't have the eventId in URl.  Should be like `/pipelines/123/events?view=grid`

**Comment by OronNadiv** [comment]
> grid view shouldn't have the eventId in URl. Should be like `/pipelines/123/events?view=grid`

I don't think that's the case.  `/pipelines/123/events` will be redirected to `/pipelines/123/events/456`.

[Here](https://github.com/screwdriver-cd/ui/blob/master/app/pipeline/events/index/route.js#L10) is the code.

File: app/pipeline/events/controller.js [hunk 1]
```diff
@@ -374,7 +380,11 @@ export default Controller.extend(ModelReloaderMixin, {
         });
       }
 
-      this.set('showListView', showListView);
+      const eventId = get(this.selectedEventObj, 'id');
+
+      this.transitionToRoute('pipeline.build', this.get('pipeline.id'), eventId, {
+        queryParams: { view: showListView ? 'list' : 'grid' }
```
Comment (adong) [inline app/pipeline/events/controller.js:386]: Is routing to another route with query params meant to keep in the browser history? So, people can go back and forward?

Comment (OronNadiv) [reply app/pipeline/events/controller.js:386]: I don't think it's necessary.  What is the best way to change the `view` query param without calling `transitionToRoute`?

Comment (adong) [reply app/pipeline/events/controller.js:386]: The old way, `this.set('view', toBe);` 

Now, you can just write, `this.view = toBe`.

Comment (OronNadiv) [reply app/pipeline/events/controller.js]: Done.  Thanks for your feedback @adong .

**Review by adong — APPROVED** [review]

**Pull request merged**